machine/v1/types_awsplacementgroup.go

```diff
@@ -0,0 +1,163 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AWSPlacementGroup ensures that a placement group matching the given configuration exists within AWS
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".spec.groupType",description="Placement Group Type"
+// +kubebuilder:printcolumn:name="Management",type="string",JSONPath=".spec.managementState",description="Management State"
+// +kubebuilder:printcolumn:name="Replicas",type="integer",JSONPath=".status.replicas",description="EC2 Replicas within the Placement Group"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="AWSPlacementGroup age"
+type AWSPlacementGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AWSPlacementGroupSpec   `json:"spec,omitempty"`
+	Status AWSPlacementGroupStatus `json:"status,omitempty"`
+}
+
+type AWSPlacementGroupSpec struct {
+	// AWSPlacementGroupManagementSpec defines the configuration for a managed or unmanaged placement group.
+	AWSPlacementGroupManagementSpec `json:",inline"`
+
+	// CredentialsSecret is a reference to the secret with AWS credentials. The secret must reside in the same namespace
+	// as the AWSPlacementGroup resourec. Otherwise, the controller will leverage the EC2 instance assigned IAM Role,
+	// in OpenShift this will always be the Control Plane Machine IAM Role.
+	// +optional
+	CredentialsSecret *LocalSecretReference `json:"credentialsSecret,omitempty"`
+}
+
+// AWSPlacementGroupManagementSpec defines the configuration for a managed or unmanaged placement group.
+// +union
+type AWSPlacementGroupManagementSpec struct {
+	// ManagementState determines whether the placement group is expected
+	// to be managed by this CRD or whether it is user managed.
+	// A managed placement group may be moved to unmanaged, however an unmanaged
+	// group may not be moved back to managed.
+	// +kubebuilder:default:=Managed
+	// +unionDiscriminator
+	// +optional
```

Comment:
A user is already going to have to set the grouptype.  I would make this required and not defaulted.